plugins/available/history-substring-search.plugin.bash

```diff
@@ -0,0 +1,9 @@
+cite about-plugin
```

Comment:
As mentioned in #1506, I think this should be one plugin + environment variables. Using two plugins will likely lead to further issues/questions down the road...

Reply:
I was thinking about that for a bit and came to the opposite conclusion. If I may make my case:

- Discoverable: I don't think the average user should have to look through documentation to discover what's going on, or look at the plugins source code. Single bash-it currently doesn't offer more than a single line description in a plug-in, specifically named plugins felt like a better trade off.

- Clarity: I came to the opposite conclusion here. This PR would make enabling history modification more explicit, since there isn't a default modification baked in, and moves search modifications to their own plugins.

- Load priority is taken care of because of lexographical ordering, giving substring search priority if both are enabled (tested locally).

Reply:
I will leave it to you guys to figure out which is the best solution. My only input is that it should be as simple and clear for end users as possible. They should not need to look at source code and it would be great if each plugin had a clear description of its purpose and any configuration options in a file, on a wiki or whatever. Perhaps even a screenshot can be included as a picture often speaks more than words.

Reply:
@cornfeedhobo Thanks for the detailed explanation of your thought process - makes sense to me. I'm fine with using your dual plugin approach. There's some code duplication in this case, but that shouldn't be much of an issue.

👍